src/typerecipes/smooth.jl

```diff
@@ -1,17 +1,20 @@
-struct Linear{S, T}
-    x::NTuple{2, S}
-    y::NTuple{2, T}
+struct GLMResult{S, T}
+    x::AbstractVector{S}
+    y::AbstractVector{T}
+    l::AbstractVector{T}
+    u::AbstractVector{T}
 end
 
-convert_arguments(P::PlotFunc, l::Linear) = PlotSpec{LineSegments}([Point2f0(x,y) for (x,y) in zip(l.x, l.y)])
-
-function linear(x, y)
-    itc, slp = hcat(fill!(similar(x), 1), x) \ y
-    xs = extrema_nan(x)
-    ys = slp .* xs .+ itc
-    Linear(xs, ys)
+function linear(x::AbstractVector{T}, y::AbstractVector{T}) where {T}
```

Comment:
```suggestion
function linear(x::AbstractVector{T}, y::AbstractVector) where {T}
```

Otherwise it fails if `x` and `y` have different eltypes (say int versus float).